### PR TITLE
New Adapter: Foreshop

### PR DESCRIPTION
**Pull request by iN-Sight**
(test endpoints)

### CODE REVIEW

**Review by guscarreon — CHANGES_REQUESTED** [review]

**Review by guscarreon — CHANGES_REQUESTED** [review]
Github actions validation is failing because of low unit-test coverage: 
```
 ok  	github.com/prebid/prebid-server/adapters/foreshop	0.024s	coverage: 3.6% of statements
Package has less than 30% code coverage. Run ./scripts/coverage.sh --html to see a detailed coverage report, and add tests to improve your coverage
```
Further coverage analysis:
```
╭─ ~/go/src/github.com/prebid/prebid-server/adapters/foreshop  ‹PR1653*›
╰─➤  go test -coverprofile=c.out
PASS
coverage: 3.6% of statements
ok      github.com/prebid/prebid-server/adapters/foreshop       0.744s

╭─ ~/go/src/github.com/prebid/prebid-server/adapters/foreshop  ‹PR1653*›
╰─➤  go tool cover -func=c.out
github.com/prebid/prebid-server/adapters/foreshop/foreshop.go:24:       Builder                 100.0%
github.com/prebid/prebid-server/adapters/foreshop/foreshop.go:32:       MakeRequests            0.0%
github.com/prebid/prebid-server/adapters/foreshop/foreshop.go:86:       MakeBids                0.0%
github.com/prebid/prebid-server/adapters/foreshop/foreshop.go:124:      getMediaType            0.0%
github.com/prebid/prebid-server/adapters/foreshop/foreshop.go:146:      getBidderData           0.0%
github.com/prebid/prebid-server/adapters/foreshop/usersync.go:11:       NewForeshopSyncer       0.0%
total:                                                                  (statements)            3.6%

File: config/config.go [hunk 1]
```diff
@@ -828,6 +828,7 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("adapters.emx_digital.endpoint", "https://hb.emxdgt.com")
 	v.SetDefault("adapters.engagebdr.endpoint", "http://dsp.bnmla.com/hb")
 	v.SetDefault("adapters.eplanning.endpoint", "http://rtb.e-planning.net/pbs/1")
+	v.SetDefault("adapters.foreshop.endpoint", "http://{{.Host}}/www/admin/api/bid-request-process.php")
```
[the remark this answers is not in the document]
Comment (SyntaxNode) [reply config/config.go:831]: It's actually a bit more than an encouragement. We will not allow adapters to pass through the host. PBS is not an open proxy.

We can guide you to a better solution. Are you using the host for debugging, vanity urls, or geo balancing?

File: adapters/foreshop/foreshop.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// Adapter hosts the Foreshop methods for making requests and bids, which are sent to its "Endpoint" address
+type Adapter struct {
```
[the remark this answers is not in the document]
Comment (SyntaxNode) [reply adapters/foreshop/foreshop.go:19]: Actually, we recommend the name "adapter" starting with a lower case letter to keep avoid exporting it.

**Comment by SyntaxNode** [comment]
We recently updated our new adapter doc: https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html

That will guide you through our rules and best practices.

**Comment by iN-Sight** [comment]
I appreciate the feedback and suggested corrections. Thanks!

File: adapters/foreshop/foreshop.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (iN-Sight) [reply adapters/foreshop/foreshop.go:19]: Noted. We currently don't actually have an endpoint for that, but I'll discuss this with the team and get that sorted.

File: config/config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (iN-Sight) [reply config/config.go:831]: Noted. We don't currently have an endpoint for this, but I'll discuss this with the team and fix the URL.

**Comment by SyntaxNode** [comment]
@iN-Sight Just wanted to check in how you're progressing. Is there anything we can assist with?

**Comment by iN-Sight** [comment]
@SyntaxNode Hello, I appreciate you checking in!
After reviewing our current structure, I've decided that we aren't ready to submit a merge request yet for the adapter. We'll submit another request at another date.

Your guidance was nonetheless very valuable though. Thanks :)

Regarding the request, I'm closing this for now. My apologies if I was supposed to do this sooner.

**Comment by SyntaxNode** [comment]
No worries. You're welcome to open a PR again when you're ready.

**Pull request closed without merging**